src/query/service/src/servers/http/v1/query/http_query_manager.rs

```diff
@@ -177,26 +170,24 @@ impl HttpQueryManager {
     }
 
     #[async_backtrace::framed]
-    pub(crate) fn remove_query(
+    pub(crate) async fn remove_query(
         self: &Arc<Self>,
         query_id: &str,
         reason: RemoveReason,
+        error: ErrorCode,
     ) -> Option<Arc<HttpQuery>> {
         // deref at once to avoid holding DashMap shard guard for too long.
         let query = self.queries.get(query_id).map(|q| q.clone());
-        query.map(|q| {
-            let not_removed_yet = !q.mark_removed(reason);
-            let to_evict = not_removed_yet
-                .then(|| {
-                    let mut queue = self.removed_queries.lock();
-                    queue.push(q.id.to_string())
-                })
-                .flatten();
-            if let Some(qid) = to_evict {
-                self.queries.remove(&qid);
+        if let Some(q) = &query {
+            if q.mark_removed(reason) {
```

Comment:
do we need to call `session.force_kill_query` here?

Reply:
it is called in `Executor::stop`